powerbi-visual-builder/src_extension/builder.ts

```diff
@@ -234,8 +246,48 @@ class PowerBIVisualGenerator implements ExportTemplateTarget {
             type = { text: true };
           }
         }
+        if (typeof property.target.property === "object" &&
```

Comment:
I'd pull this out into a variable, `const targetProperty = property.target.property`, and replace all of the below instances of `property.target.property`, it makes it easier to read (for me at least)